posthog/client.py

```diff
@@ -162,6 +166,165 @@ def sync_execute(query, args=None, settings=None, with_column_types=False):
     return result
 
 
+REDIS_STATUS_TTL = 600  # 10 minutes
+
+
+@dataclass_json
+@dataclass
+class QueryStatus:
+    team_id: int
+    num_rows: float = 0
+    total_rows: float = 0
+    status: str = "submitted"
+    complete: bool = False
+    error: bool = False
```

Comment:
Having `status`, `complete`, and `error` feels a bit odd because it's the same information in 2 different ways. I'm wondering if we have only `status`?